version : '2'
services:
  auth:
    image: pdns_auth
    # deploy:
    #   replicas: 2
    #   restart_policy:
    #     condition: on-failure
    volumes:
      - ./Config/rsyslogd.conf:/etc/rsyslogd.conf
      - ./Config/pdns.conf:/etc/pdns/pdns.conf
      - ./Logs/auth:/var/log/
    ports:
      - "5300:5300/udp"
      - "5300:5300/tcp"
    networks:
      dns:
        ipv4_address: "172.18.0.100"

  dnsdist:
    image: pdns_dnsdist
    # deploy:
    #   replicas: 2
    #   restart_policy:
    #     condition: on-failure
    volumes:
      - ./Config/rsyslogd.conf:/etc/rsyslogd.conf
      - ./Config/dnsdist.conf:/etc/dnsdist/dnsdist.conf
      - ./Logs/dnsdist:/var/log/
    ports:
      - "5301:5301/udp"
      - "5301:5301/tcp"
      - "8083:8083/tcp"
    networks:
      dns:
        ipv4_address: "172.18.0.101"

  recursor:
    image: pdns_recursor
    # deploy:
    #   replicas: 2
    #   restart_policy:
    #     condition: on-failure
    volumes:
      - ./Config/rsyslogd.conf:/etc/rsyslogd.conf
      - ./Config/recursor.conf:/etc/pdns-recursor/recursor.conf
      - ./Config/root.hints:/etc/pdns-recursor/root.hints
      - ./Logs/recursor:/var/log/
    ports:
      - "5302:5302/udp"
      - "5302:5302/tcp"
    networks:
      dns:
        ipv4_address: "172.18.0.102"

networks:
  dns:
    driver: bridge
    ipam:
     config:
       - subnet: "172.18.0.0/24"
         gateway: "172.18.0.1"
         